eclipse-theia/theia-blueprint
# Theia 1.36.0, enable differential updater for windows, use vscode.git [ jenkins ]

#### What it does

Updates Theia and dependencies to the latest version and starts to use the vscode.git extension again. 

Moreover this enables differential updates for windows. The Updater will use the blockmap (see https://download.eclipse.org/theia/latest/windows/) to compute a diff and only download changed blocks instead of the full .exe

For this we need to update `electron-updater` to a newer version to get a required bugfix and to adjust the main URL where the windows installer is published. This is required so that older versions may still find their own blockmap besides the latest one.
Moreover we have to update the `latest.yml` files for older windows versions accordingly when a new version is released. 

Since the eclipse foundation offers a web service for signing executables, we can't use electron-builders built-in signing support. Therefore we recompute the blockmap manually after we have signed the installer. 

Contributed on behalf of STMicroelectronics

Signed-off-by: Johannes Faltermeier <jfaltermeier@eclipsesource.com>

#### How to test
Build and check whether Blueprint and Git Support is still working. 

Differential updates are hard to test without actually running the build pipeline. 

Locally you may adjust `electron-builder.yml` and adjust `"https://download.eclipse.org/theia/${version}/windows"` to some localhost path. Then you may build a few version (see the `Jenkinsfile`) and use `npx serve -p 5000` to run a local file server. 

I've did some builds in this location: https://download.eclipse.org/theia/diff-update-test/1.34.1/windows/
This application will update to 1.35.0 using differential updates. 

#### Review checklist


#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
